provisioning/baremetal_secrets.go

```diff
@@ -168,3 +154,41 @@ password = %s
 	_, err = client.Secrets(targetNamespace).Create(context.Background(), secret, metav1.CreateOptions{})
 	return err
 }
+
+func CreateAllSecrets(client coreclientv1.SecretsGetter, targetNamespace string, baremetalConfig *metal3iov1alpha1.Provisioning, scheme *runtime.Scheme) error {
+	// Create a Secret for the Mariadb Password
+	if err := createMariadbPasswordSecret(client, targetNamespace, baremetalConfig, scheme); err != nil {
+		return errors.Wrap(err, "failed to create Mariadb password")
+	}
+	// Create a Secret for the Ironic Password
+	if err := createIronicSecret(client, targetNamespace, ironicSecretName, ironicUsername, "ironic", baremetalConfig, scheme); err != nil {
+		return errors.Wrap(err, "failed to create Ironic password")
+	}
+	// Create a Secret for the Ironic RPC Password
+	if err := createIronicSecret(client, targetNamespace, ironicrpcSecretName, ironicrpcUsername, "json_rpc", baremetalConfig, scheme); err != nil {
+		return errors.Wrap(err, "failed to create Ironic rpc password")
+	}
+	// Create a Secret for the Ironic Inspector Password
+	if err := createIronicSecret(client, targetNamespace, inspectorSecretName, inspectorUsername, "inspector", baremetalConfig, scheme); err != nil {
+		return errors.Wrap(err, "failed to create Inspector password")
+	}
+	return nil
+}
+
+func DeleteAllSecrets(info *ProvisioningInfo) []string {
+	deleteFailures := []string{}
+	if err := info.Client.CoreV1().Secrets(info.Namespace).Delete(context.Background(), baremetalSecretName, metav1.DeleteOptions{}); err != nil {
+		deleteFailures = append(deleteFailures, baremetalSecretName)
+	}
+	if err := info.Client.CoreV1().Secrets(info.Namespace).Delete(context.Background(), ironicSecretName, metav1.DeleteOptions{}); err != nil {
+		deleteFailures = append(deleteFailures, ironicSecretName)
+	}
+	if err := info.Client.CoreV1().Secrets(info.Namespace).Delete(context.Background(), inspectorSecretName, metav1.DeleteOptions{}); err != nil {
+		deleteFailures = append(deleteFailures, inspectorSecretName)
+	}
+	if err := info.Client.CoreV1().Secrets(info.Namespace).Delete(context.Background(), ironicrpcSecretName, metav1.DeleteOptions{}); err != nil {
+		deleteFailures = append(deleteFailures, ironicrpcSecretName)
+	}
+
+	return deleteFailures
```

Comment:
return errors.NewAggregate(errs)

Reply:
Didn't want to import a new errors package so achieved the same result differently.